instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -49,9 +49,16 @@ def clear_cached_config
 
           def initialize(app)
             @app = app
+            @untraced_endpoints = config[:untraced_endpoints].is_a?(Array) ? config[:untraced_endpoints] : []
```

Comment:
Some sort of warning here would be nice, if `config[:untraced_endpoints]` is present and not an array.

Of course, there's a question of whether we want to be so prescriptive about the type -- it seems like anything supporting `.include?` should be fine, e.g. if someone wanted to wrap a regex.

Reply:
Anything supporting `include?` seems potentially risky.  If for example someone just passed in `/`, most endpoints would go untraced.

What if we expected a lambda to be provided that accepts the `env`, and that determined if it should be untraced or not?  That seems a bit more flexible as someone could making tracing decisions based off of headers, path, etc...

Reply:
> If for example someone just passed in `/`, most endpoints would go untraced.

That's not entirely true 😄 
```
irb(main):001:0> '/'.include? '/foo'
=> false
```

There's a trade-off between flexibility and complexity of use. We should at least make the most common use-cases easy (e.g. health checks, for us). Path seems like the most common thing to want to match. I didn't see anything that would let us "duck type" regexes and arrays 🤷 .

I'm not terribly fussed - if you want to punt the more complex use-cases and can support that without changing the interface, I'll 👍

Reply:
>That's not entirely true 😄

I may or may not have been thinking of it in reverse `path.include?('/')` 🤦 

>There's a trade-off between flexibility and complexity of use. We should at least make the most common use-cases easy (e.g. health checks, for us).

I agree, definitely want to keep it simple.  Trying to think ahead a little bit I suspect we'll want to support a few use-cases.

- Allow the user to specify one or many endpoints
- Allow the user to supply those endpoints via env var
- Ability to make tracing decisions off of something other than the endpoint (headers)

I think I can come up with a simple interface that supports these scenarios.

Reply:
>Some sort of warning here would be nice, if config[:untraced_endpoints] is present and not an array.

Also regarding this comment, I've been thinking of this a little bit more and wondering if we should have some config checking on install/initialization instead of dealing with it at runtime.

Reply:
To keep this interface simple, I'd like to keep this as expecting and only accepting an array of string values.  I'll do the config validation as a separate PR from this one.